src/docs/development/platform-integration/add-to-app-android/add-splash-screen.md

```diff
@@ -0,0 +1,221 @@
+a `FlutterActivity`. Here are instructions to display an Android launch screen:
+
+### Define a launch theme
+
+In `styles.xml`, define a theme whose `windowBackground` is set to the
+`Drawable` that should be displayed as the launch screen.
+
+```xml
+<style name="LaunchTheme" parent="@android:style/Theme.Black.NoTitleBar">
+    <item name="android:windowBackground">@drawable/launch_background</item>
+</style>
+```
+
+### Define a normal theme
+
+In `styles.xml`, define a normal theme to be applied to `FlutterActivity` after
+the launch screen is gone. The normal theme background only shows for a very
+brief moment after the splash screen disappears, and during orientation change
+and `Activity` restoration. Therefore, it is recommended that the normal theme
+use a solid background color that looks similar to the primary background color
+of the Flutter UI.
+
+```xml
+<style name="NormalTheme" parent="@android:style/Theme.Black.NoTitleBar">
+    <item name="android:windowBackground">@drawable/normal_background</item>
+</style>
+```
+
+### Setup FlutterActivity in AndroidManifest.xml
+
+In `AndroidManifest.xml`, set the `theme` of `FlutterActivity` to the launch
+theme. Then, add a meta-data element to the desired `FlutterActivity` to instruct
+Flutter to switch from the launch theme to normal theme at the appropriate time.
+
+```xml
+<activity
+    android:name=".MyActivity"
+    android:theme="@style/LaunchTheme"
+    // ...
+    >
+    <meta-data
+        android:name="io.flutter.embedding.android.NormalTheme"
+        android:resource="@style/NormalTheme"
+        />
+    <intent-filter>
+        <action android:name="android.intent.action.MAIN"/>
+        <category android:name="android.intent.category.LAUNCHER"/>
+    </intent-filter>
+</activity>
+```
+
+The Android app now displays the desired launch screen while the app
+initializes.
+
+## Flutter Splash Screen
+
+Each Flutter experience in an app requires a few moments to initialize the Dart
+isolate that runs the code. This means a user momentarily sees a blank screen.
+Flutter supports an improved user experience by displaying an Android `View` as 
+a splash screen while Flutter initializes.
+
+Flutter supports two options for a splash screen. Developers can choose to
+display a `Drawable` of their choice, which then fades out once initialization
+is complete. The other option is to provide a custom `SplashScreen`, which is
+capable of displaying any Android `View` content that is desired.
+
+### Showing a Drawable splash screen
+
+A `Drawable` splash screen can be configured for a `FlutterActivity`,
+`FlutterFragment`, or `FlutterView`.
+
+#### In a FlutterActivity
+
+To display a `Drawable` as a Flutter splash screen in a `FlutterActivity`, add
+the following meta-data to the associated `FlutterActivity` in
+`AndroidManifest.xml`.
+
+```xml
+<meta-data
+    android:name="io.flutter.embedding.android.SplashScreenDrawable"
+    android:resource="@drawable/my_splash"
```

Comment:
Should this be by default (or otherwise document that they can): make this drawable the same drawable that we set as the LaunchTheme windowBackground's drawable (something most people will want to do I assume)?

Reply:
done